src/CodeGen_WebAssembly.cpp

```diff
@@ -110,13 +125,114 @@ void CodeGen_WebAssembly::init_module() {
     }
 }
 
-string CodeGen_WebAssembly::mcpu() const {
+void CodeGen_WebAssembly::codegen_vector_reduce(const VectorReduce *op, const Expr &init) {
+#if LLVM_VERSION >= 130
+    struct Pattern {
+        VectorReduce::Operator reduce_op;
+        int factor;
+        Expr pattern;
+        const char *intrin;
+        Type narrow_type;
```

Comment:
I don't think you need this, all of the patterns have the narrow type the same as the pattern's type.

Reply:
At the moment, yes, but I'm not sure if that will be the case for everything, and I wanted to preserve this logic (adapted from Codegen_X86.cpp). That said, easy enough to restore if/when it's needed.

Reply:
I think we should just add it when needed. It adds a fair bit of code that is currently dead. Note that CodeGen_ARM doesn't have either of these.